include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -6132,8 +6154,7 @@ struct OrtApi {
    * \since Version 1.23
    */
   ORT_API2_STATUS(CreateSharedAllocator, _In_ OrtEnv* env, _In_ const OrtEpDevice* ep_device,
-                  _In_ OrtDeviceMemoryType mem_type, _In_ OrtAllocatorType allocator_type,
-                  _In_opt_ const OrtKeyValuePairs* allocator_options,
+                  _In_ OrtDeviceMemoryType mem_type, _In_opt_ const OrtKeyValuePairs* allocator_options,
```

Comment:
This change was to keep the OrtArenaAllocator as an internal only type.

But there's also the new read-only allocator from #25348. Is there a valid use case where a user would need/want to use the read-only allocator to create on-device shared weights outside of an inference session. If so, leaving the OrtAllocatorType in here would be required.